Build failure: llvmPackages_14.llvm [armv7l-linux]

### Steps To Reproduce

Steps to reproduce the behavior:
1. build llvm on https://github.com/NixOS/aarch64-build-box like this `nix-build . -A pkgs.llvmPackages_14.llvm --system armv7l-linux`

### Build log
Broken on 14, 15, 16, 17, 18 and git versions.

llvmPackages_14.llvm:
```
********************
Failed Tests (2):
  LLVM :: tools/llvm-dwarfdump/X86/prettyprint_types.s
  LLVM :: tools/llvm-dwarfdump/X86/simplified-template-names.s


Testing Time: 83.17s
  Skipped          :     9
  Unsupported      :  1417
  Passed           : 46582
  Expectedly Failed:   184
  Failed           :     2
make[3]: *** [CMakeFiles/check-all.dir/build.make:71: CMakeFiles/check-all] Error 1
```

llvmPackages_15.llvm:
```
********************
Failed Tests (4):
  LLVM :: CodeGen/RISCV/rv32zbp.ll
  LLVM :: CodeGen/RISCV/rv64zbp.ll
  LLVM :: tools/llvm-dwarfdump/X86/prettyprint_types.s
  LLVM :: tools/llvm-dwarfdump/X86/simplified-template-names.s


Testing Time: 102.47s
  Skipped          :    20
  Unsupported      :  1693
  Passed           : 48495
  Expectedly Failed:   183
  Failed           :     4
FAILED: CMakeFiles/check-all /build/llvm-src-15.0.7/llvm/build/CMakeFiles/check-all
```

llvmPackages_16.llvm:
```
********************
Failed Tests (2):
  LLVM :: tools/llvm-dwarfdump/X86/prettyprint_types.s
  LLVM :: tools/llvm-dwarfdump/X86/simplified-template-names.s


Testing Time: 112.96s
  Skipped          :    20
  Unsupported      :  1907
  Passed           : 51481
  Expectedly Failed:   182
  Failed           :     2
FAILED: CMakeFiles/check-all /build/llvm-src-16.0.6/llvm/build/CMakeFiles/check-all
```

llvmPackages_17.llvm:
```
********************
Failed Tests (2):
  LLVM :: CodeGen/RISCV/attributes.ll
  LLVM :: CodeGen/RISCV/xtheadmempair.ll


Testing Time: 88.08s
  Skipped          :    35
  Unsupported      :  1930
  Passed           : 53865
  Expectedly Failed:   183
  Failed           :     2
FAILED: CMakeFiles/check-all /build/llvm-src-17.0.6/llvm/build/CMakeFiles/check-all
```

llvmPackages_18.llvm:
```
********************
Failed Tests (7):
  LLVM :: CodeGen/RISCV/attributes.ll
  LLVM :: CodeGen/RISCV/rvv/combine-store-extract-crash.ll
  LLVM :: CodeGen/RISCV/xtheadmempair.ll
  LLVM :: tools/dsymutil/X86/DWARFLinkerParallel/odr-fwd-declaration3.test
  LLVM :: tools/dsymutil/X86/DWARFLinkerParallel/odr-predictable-output2.test
  LLVM :: tools/dsymutil/X86/DWARFLinkerParallel/odr-string.test
  LLVM :: tools/dsymutil/X86/DWARFLinkerParallel/odr-uniquing.cpp


Testing Time: 97.19s

Total Discovered Tests: 59250
  Skipped          :    36 (0.06%)
  Unsupported      :  2101 (3.55%)
  Passed           : 56928 (96.08%)
  Expectedly Failed:   178 (0.30%)
  Failed           :     7 (0.01%)
FAILED: CMakeFiles/check-all /build/llvm-src-18.1.5/llvm/build/CMakeFiles/check-all
```

llvmPackages_git.llvm:
```
********************
Failed Tests (9):
  LLVM :: CodeGen/RISCV/rvv/combine-store-extract-crash.ll
  LLVM :: CodeGen/RISCV/xtheadmempair.ll
  LLVM :: tools/dsymutil/ARM/DWARFLinkerParallel/accel-imported-declarations.test
  LLVM :: tools/dsymutil/X86/DWARFLinkerParallel/keep-func.test
  LLVM :: tools/dsymutil/X86/DWARFLinkerParallel/odr-fwd-declaration2.test
  LLVM :: tools/dsymutil/X86/DWARFLinkerParallel/odr-nested-types2.test
  LLVM :: tools/dsymutil/X86/DWARFLinkerParallel/odr-predictable-output2.test
  LLVM :: tools/dsymutil/X86/DWARFLinkerParallel/odr-string.test
  LLVM :: tools/llvm-dwarfutil/ELF/X86/dwarf5-loclists.test


Testing Time: 97.74s

Total Discovered Tests: 60272
  Skipped          :    36 (0.06%)
  Unsupported      :  2293 (3.80%)
  Passed           : 57756 (95.83%)
  Expectedly Failed:   178 (0.30%)
  Failed           :     9 (0.01%)
FAILED: CMakeFiles/check-all /build/llvm-src-19.0.0-unstable-2024-04-07/llvm/build/CMakeFiles/check-all
```

### Additional context

Likely broken by https://github.com/NixOS/nixpkgs/pull/303521

See also https://github.com/llvm/llvm-project/issues/62195 https://github.com/llvm/llvm-project/issues/91335

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@RossComputerGuy 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-linux"`
 - host os: `Linux 5.15.156, NixOS, 24.05 (Uakari), 24.05.20240418.bc279bb`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/run/current-system/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/compilers/llvm/common/llvm/default.nix
+++ b/pkgs/development/compilers/llvm/common/llvm/default.nix
@@ -28,7 +28,7 @@
 , sysctl
 , buildLlvmTools
 , debugVersion ? false
-, doCheck ? (if lib.versionOlder release_version "15" then stdenv.isLinux else true)
+, doCheck ? !stdenv.isAarch32 && (if lib.versionOlder release_version "15" then stdenv.isLinux else true)
   && (!stdenv.isx86_32 /* TODO: why */) && (!stdenv.hostPlatform.isMusl)
   && (stdenv.hostPlatform == stdenv.buildPlatform)
 , enableManpages ? false

